app/jobs/checksum_report_job.rb

```diff
@@ -22,23 +22,35 @@ def perform(bulk_action_id, params)
       update_druid_count
       begin
         raise "#{Time.current} ChecksumReportJob not authorized to view all content}" unless ability.can?(:view_content, ActiveFedora::Base)
-
-        response = conn.post '/objects/checksums', druids: pids, format: 'csv'
-        unless response.success? # an internal API call failing is significant enough to notify HB specifically over as well as failing the job
-          message = "Call to preservation_catalog_api failed: #{response.status} #{response.reason_phrase}: #{response.body}"
-          Honeybadger.notify message
-          raise message
-        end
-        File.open(report_filename, 'w') { |file| file.write(response.body) }
+        csv_report = Preservation::Client.objects.checksums(druids: pids)
+        File.open(report_filename, 'w') { |file| file.write(csv_report) }
         bulk_action.update(druid_count_success: pids.length) # this whole job is run in one call, so it either all succeeds or fails
+      rescue Preservation::Client::NotFoundError, Preservation::Client::UnexpectedResponseError => e
+        bulk_action.update(druid_count_fail: pids.length)
+        message = "#{Time.current} ChecksumReportJob got error from Preservation Catalog API: #{e.class} #{e.message}"
+        log.puts(message)
+        # honeybadger and other notifications should happen at prescat level
+      rescue Preservation::Client::ConnectionFailedError => e
+        bulk_action.update(druid_count_fail: pids.length)
+        message = "#{Time.current} ChecksumReportJob failed on call to Preservation Catalog API: #{e.class} #{e.message}"
+        log.puts(message)
+        # if we couldn't connect to preservation API, we have an issue
+        Honeybadger.notify message
+        raise message
       rescue StandardError => e
         bulk_action.update(druid_count_fail: pids.length)
-        log.puts("#{Time.current} ChecksumReportJob creation failed #{e.class} #{e.message}")
+        message = "#{Time.current} ChecksumReportJob creation failed #{e.class} #{e.message}"
+        log.puts(message)
+        # if we couldn't write out the file, we have an issue
+        Honeybadger.notify message
+        raise message
```

Comment:
Is this where we end up if user is disallowed?  If so, should we be more specific in the error we're trapping for, and NOT raise the exception?

I thought this was catching for problems opening file.

Reply:
i'm not sure if this is where unauthorized users end up actually -- i was more thinking it was a just a general catch all ... but we could just let it go for now and see what it catches (and if it catches a bunch of authorization errors, extend to improve messaging)